TST: ASDF test_isot fails with AssertionError in dev job

Example log: https://github.com/astropy/astropy/runs/6908150791?check_suite_focus=true

It ran with `asdf: 2.12.1.dev26+g7e34de4`.

```
__________________________________ test_isot ___________________________________
tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_isot0')
    def test_isot(tmpdir):
        tree = {
            'time': time.Time('2000-01-01T00:00:00.000')
        }
        helpers.assert_roundtrip_tree(tree, tmpdir)
        ff = asdf.AsdfFile(tree)
        tree = yamlutil.custom_tree_to_tagged_tree(ff.tree, ff)
>       assert isinstance(tree['time'], str)
E       AssertionError: assert False
E        +  where False = isinstance({'value': '2000-01-01T00:00:00.000', 'base_format': 'isot'}, str)
.../astropy/io/misc/asdf/tags/time/tests/test_time.py:92: AssertionError
```

The dev job [succeeded in main 4 days ago](https://github.com/astropy/astropy/runs/6852052360?check_suite_focus=true) with `asdf: 2.12.1.dev21+g5bfe173`.

@WilliamJamieson , any idea what happened and how to fix? Thanks!

--- a/tox.ini
+++ b/tox.ini
@@ -85,9 +85,9 @@ deps =
 
     # The devdeps factor is intended to be used to install the latest developer version
     # or nightly wheel of key dependencies.
+    # Do not install asdf-astropy so we can test astropy.io.misc.asdf until we remove it.
     devdeps,mpldev: git+https://github.com/matplotlib/matplotlib.git#egg=matplotlib
     devdeps: git+https://github.com/asdf-format/asdf.git#egg=asdf
-    devdeps: git+https://github.com/astropy/asdf-astropy.git#egg=asdf-astropy
     devdeps: git+https://github.com/liberfa/pyerfa.git#egg=pyerfa
 
 # The following indicates which extras_require from setup.cfg will be installed

